### PR TITLE
feat(rag): add project-scoped indexes

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Add a `project` parameter to `rag_index`, `rag_search`, and `rag_status` that isolates each project's RAG index to its own directory under `XDG_DATA_HOME/gptme/rag/{project}/`
- When `project` is set, `--persist-dir` is passed to `gptme-rag` so indexed documents never pollute searches in other projects or the global index
- Fully backward compatible: when `project` is omitted, behaviour is identical to before
- Project names with slashes, spaces, or special characters are sanitized to a filesystem-safe form

## Problem solved

Before this change, `rag_index` and `rag_search` used a single global index. Indexing docs for one project would pollute search results in every other project, making RAG unreliable in multi-repo workflows.

## Implementation notes

- `rag_index` and `rag_search` pass `--persist-dir <path>` to the CLI (supported by gptme-rag)
- `rag_status` uses directory introspection for project-scoped status (the CLI's `status` subcommand doesn't support `--persist-dir`)
- 17 new tests covering sanitization, persist-dir routing, backward compat, and isolation invariant

## Test plan

- [ ] `uv run pytest tests/test_tools_rag.py -q` — all tests pass
- [ ] `mypy gptme/tools/rag.py` — clean
- [ ] Verify `rag_index(project="A")` + `rag_search("query")` does NOT return project A docs (isolation)

### CODE REVIEW

File: gptme/tools/rag.py [hunk 1]
```diff
@@ -51,13 +52,30 @@
...
 logger = logging.getLogger(__name__)
 
+
+def _sanitize_project_name(name: str) -> str:
+    """Sanitize a project name for use as a filesystem path component."""
+    # Replace path separators and whitespace with underscores
+    sanitized = re.sub(r"[/\\\s]+", "_", name)
+    # Remove any remaining non-alphanumeric/underscore/hyphen/dot characters
+    sanitized = re.sub(r"[^\w\-.]", "_", sanitized)
+    return sanitized.strip("_") or "default"
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/tools/rag.py]: Fixed in 19db35185: changed  to call  (stripping both underscores and dots) so `." → `"default"` and ".." → `"default"`, while internal dots in names like `v1.2.3` or `my.project` are preserved. Added tests for these two edge cases.

File: gptme/tools/rag.py [hunk 2]
```diff
@@ -156,33 +174,83 @@ def _run_rag_cmd(cmd: list[str]) -> subprocess.CompletedProcess:
...
+            global index status.
+    """
+    persist_dir = _project_persist_dir(project)
+    if persist_dir is not None:
+        # gptme-rag status does not support --persist-dir, so we report directory info
+        if persist_dir.exists():
+            file_count = sum(1 for f in persist_dir.rglob("*") if f.is_file())
+            size_bytes = sum(
+                f.stat().st_size for f in persist_dir.rglob("*") if f.is_file()
+            )
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/tools/rag.py]: Fixed in 19db35185: collect rglob results into a list once, then use len() and sum() over that list — avoids the double filesystem traversal and eliminates the inconsistency window.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
@greptileai review
